internal/versions/versions.go

```diff
@@ -175,42 +171,88 @@ func (h *handler) GetDefaultReleaseImage(cpuArchitecture string) (*models.Releas
 
 // Returns the OsImage entity
 func (h *handler) GetOsImage(openshiftVersion, cpuArchitecture string) (*models.OsImage, error) {
-	versionKey, err := h.getKey(openshiftVersion)
-	if err != nil {
-		return nil, err
-	}
-
-	for _, osImage := range h.osImages {
+	// Find OS images of relevant CPU architecture
+	osImages := funk.Filter(h.osImages, func(osImage *models.OsImage) bool {
+		// Empty implies default CPU architecture
+		if swag.StringValue(osImage.CPUArchitecture) == "" {
+			return cpuArchitecture == "" || cpuArchitecture == common.DefaultCPUArchitecture
+		}
 		if cpuArchitecture == "" {
-			// Empty implies default CPU architecture
 			cpuArchitecture = common.DefaultCPUArchitecture
```

Comment:
Can this be done before the `funk.Filter` call?
It seems strange to edit this variable inside the filter function.

Reply:
I think this would also simplify the return in the previous block.

Reply:
Right, will simplfy.